### PR TITLE
workflows: update to have 1.16 in some of the workflows 

### PR DESCRIPTION
**Pull request by nfuden**
Make a placeholder for 1.17 to make sure that we dont mess up on 1.16 targeted things. 
Also update some of our workflows to be 1.16 aware

### CODE REVIEW

File: docs/active_versions.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "latest": "v1.15.x",
```
Comment (inFocus7) [inline docs/active_versions.json:2]: are we planning on updating this after the first non-rc release of `1.16`?

File: .github/workflows/docs-gen.yaml [hunk 2]
```diff
@@ -7,6 +7,7 @@ on:
   push:
     branches:
     - 'main'
+    - 'v1.16.x'
     - 'v1.15.x'
     - 'v1.14.x'
     - 'v1.13.x'
```
Comment (inFocus7) [inline .github/workflows/docs-gen.yaml:13]: Will we want to remove `v1.12.x` from our docs as well?

**Review by inFocus7 — COMMENTED** [review]
a few nit comments. 

also noting that we may want to update `DOCS-ISSUE.yml` so latest is 1.16.x, and `main` references `1.17.x beta`. although this could wait til after the official release of 1.16.0

File: .github/workflows/docs-gen.yaml (hunk 2 above)
Comment (nfuden) [reply .github/workflows/docs-gen.yaml:13]: I erred on the side of not removing 1.12 until we hit ga

File: docs/active_versions.json (hunk 1 above)
Comment (nfuden) [reply docs/active_versions.json:2]: yes

File: .github/workflows/docs-gen.yaml (hunk 2 above)
Comment (inFocus7) [reply .github/workflows/docs-gen.yaml:13]: makes sense! 👍🏼 technically it's still in our support until 1.16's release, so we could hold off on removing it

**Review by inFocus7 — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**